test/aia-test-srv/main.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"regexp"
+	"time"
+
+	"github.com/letsencrypt/boulder/cmd"
+	"github.com/letsencrypt/boulder/issuance"
+)
+
+type aiaTestSrv struct {
+	issuersByName map[string]*issuance.Certificate
+}
+
+func (srv *aiaTestSrv) handleIssuer(w http.ResponseWriter, r *http.Request) {
+	issuerName, err := url.PathUnescape(r.URL.Path[1:])
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	issuer, ok := srv.issuersByName[issuerName]
+	if !ok {
+		w.WriteHeader(http.StatusForbidden) // TODO make this StatusNotFound
+		w.Write([]byte(fmt.Sprintf("issuer %q not found", issuerName)))
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/pkix-cert")
+	w.WriteHeader(http.StatusOK)
+	w.Write(issuer.Certificate.Raw)
+}
+
+// This regex excludes the "...-cross.cert.pem" files, since we don't serve our
+// cross-signed certs at AIA URLs.
+var issuerCertRegex = regexp.MustCompile(`int-(rsa|ecdsa)-[a-z]\.cert\.pem$`)
```

Comment:
File name structure holding being significant could be surprising to someone down the line when they go to tweak the generate code or config. I'm not sure what the right solution is but noting this in the hierarchy README.md might be a good place to start. Potentially a test which fails if a file with an unexpected naming convention finds itself in the hierarchy dir? I'm not sure.

Reply:
Ah, another idea here is to just have this fail to start if there's a hierarchy file that's not following your convention.

Reply:
I've added a note to the generation script where these file names are created.

Erroring out on unrecognized files would be hard, because there are a lot of other files in the hierarchy dir: pubkeys, pkcs11 configs, CRLs, etc.